### PR TITLE
[gitlab] Only run macOS unit tests on full pipelines

### PR DESCRIPTION
**Pull request by KSerrania**
### What does this PR do?

Changes the rules associated with the macOS unit tests to only run on full pipelines and main / release branches.

### Motivation

Because of org-wide resource constraints, we need to reduce the amount of macOS workflows that we run. This attempt tries to remove unit tests and keep linters, because:
- linters run faster than unit tests
- linters will still catch compilation errors (eg. variables / functions that are not defined on macOS)
- unit tests do still run on Linux on Windows, so the "only" thing we lose are unit tests that would behave differently on macOS. This does happen (so we'll get more failures reaching main), but is relatively rare.

### Additional Notes

n/a

### Possible Drawbacks / Trade-offs

Some macOS-specific failures won't be caught in PRs, and will reach main.

### Describe how to test/QA your changes

Check PR pipeline: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/29961756

### CODE REVIEW

**Review by spencergilbert — APPROVED** [review]

File: .gitlab/source_test/macos.yml [hunk 1]
```diff
@@ -27,7 +28,7 @@ tests_macos:
 
 lint_macos:
   stage: source_test
-  rules: !reference [.on_a6]
+  rules: !reference [.on_a7]
```
Comment (amenasria) [inline .gitlab/source_test/macos.yml:31]: **❓ question:** ‏What is this change for ?

**Review by amenasria — APPROVED** [review]
LGTM !

File: .gitlab/source_test/macos.yml (hunk 1 above)
Comment (KSerrania) [reply .gitlab/source_test/macos.yml:31]: Future-proofing - I don't want us to accidentally stop linting on macOS the day we stop building Agent 6 by default (if that ever happens).

**Pull request merged**